src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,137 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROW_CAPACITY_COEFFICIENT = 1.5;
+    private Object[] data;
+    private int size;
+
+    public ArrayList() {
+        this.data = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size >= data.length) {
+            data = grow();
+        }
+        data[size] = value;
+        size += 1;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Index "
+                    + index
+                    + " is out of bounds for the size "
+                    + (size));
+        }
+        isIndexPositiveNumber(index);
+        if (data.length < size + 1) {
+            data = grow();
+        }
+        System.arraycopy(data, index, data, index + 1, size - index);
+        data[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        int newSize = this.size + list.size();
+        if (data.length < newSize) {
+            data = grow(newSize);
+        }
+        for (int i = size, j = 0; i < newSize; i++, j++) {
+            data[i] = list.get(j);
+        }
+        size += list.size();
     }
 
     @Override
+    @SuppressWarnings("unchecked")
     public T get(int index) {
-        return null;
+        isIndexInBounds(index);
+        isIndexPositiveNumber(index);
```

Comment:
create method checkIndex 
where `if (index < 0 || index >= size)`

```suggestion
        checkIndex(index);
```

Reply:
Can you please suggest how to better implement checkIndex(). Can i have two separate if blocksin the method: first with i < 0, second with i >= size, because i need to throw different exceptions for these cases?

Reply:
@Elena-Bruyako Also why is it better to perform two different checks in one method?

Reply:
@Elena-Bruyako Can you please suggest how to implement this check. As for now i applied all the changes you requested(did not push because some tests are failing).
I've added checkIndex method, but if i use `(index < 0 || index >= size)` i get two tests fail with:
`core.basesyntax.ArrayListIndexOutOfBoundsException: Index 0 is out of bounds for the size 0
core.basesyntax.ArrayListIndexOutOfBoundsException: Index 5 is out of bounds for the size 5`
but if i use `(index < 0 || index > size)` condition i get three tests fail with:
`java.lang.AssertionError: Expected exception: core.basesyntax.ArrayListIndexOutOfBoundsException
java.lang.Exception: Unexpected exception, expected<core.basesyntax.ArrayListIndexOutOfBoundsException> but was<java.lang.ArrayIndexOutOfBoundsException>`
Can't get what am i doing wrong.